### ISSUE TITLE
Table object is 'false' when baking underscore separated table

### ISSUE THREAD

**Issue by GusAntoniassi** [cakephp/cakephp#10948]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.4.11
* Platform and Target: 
Apache/2.4.26 (Win32) OpenSSL/1.0.2l PHP/7.1.7
Windows NT 10.0 build 15063 (Windows 10) i586
libmysql - mysqlnd 5.0.12-dev - 20150407
Running under XAMPP v3.2.2

### What you did

1. Started a fresh project with `composer self-update && composer create-project --prefer-dist cakephp/app {project}`
2. Created the following test database:
```
CREATE TABLE `underscore_tests` (
  `id` int(11) NOT NULL,
  `col` varchar(255) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
ALTER TABLE `underscore_tests`
  ADD PRIMARY KEY (`id`);
ALTER TABLE `underscore_tests`
  MODIFY `id` int(11) NOT NULL AUTO_INCREMENT;
```
3. Updated `app.php` with database configuration data
4. Ran bake with `cake bake all underscore_tests`

### What happened
On `UnderscoreTestsController.php`, object `$this->UnderscoreTests` has the value of `false`. Trying to access `localhost/{project}/underscoretests` gives the error "Unable to locate an object compatible with paginate.", likely because of the query `$this->paginate($this->UnderscoreTests);` with `$this->UnderscoreTests` having the value of `false`.
Trying to access the object `$this->Underscoretests` works fine and debugging it returns the following:
```
object(App\Model\Table\UnderscoreTestsTable) {

	'registryAlias' => 'Underscoretests',
	'table' => 'underscore_tests',
	'alias' => 'Underscoretests',
	'entityClass' => 'App\Model\Entity\UnderscoreTest',
	'associations' => [],
	'behaviors' => [],
	'defaultConnection' => 'default',
	'connectionName' => 'default'

}
```
I am not sure if this is a problem with CakePHP's core or the Bake utility. Running `cake bake all` on console returns me:
```
Welcome to CakePHP v3.4.11 Console
---------------------------------------------------------------
App : src
Path: C:\xampp\htdocs\cake-bug\cake-bug\src\
PHP : 7.1.7
---------------------------------------------------------------
Bake All
---------------------------------------------------------------
Possible model names based on your database:
- underscore_tests
Run `cake bake all [name]` to generate skeleton files.
```
So I am sure I am using the correct name.
If I try to access `localhost/{project}/underscore_tests` I get a `MissingControllerException`, telling me that the file `src\Controller\UnderscoreTestsController.php` does not exists, which is not true since `dir src\Controller\UnderscoreTestsController.php` returns me:
```
[...]
 Pasta de C:\xampp\htdocs\{project}\src\Controller

25/07/2017  23:22             3.728 UnderscoreTestsController.php
               1 arquivo(s)          3.728 bytes
               0 pasta(s)   905.965.699.072 bytes disponíveis
```

### What you expected to happen
I would expect either `bake` to generate the table object names like `$this->Underscoretests` or be able to access `localhost/{project}/underscore_tests` without having a `MissingControllerException`.

I am available to provide more information if needed.

**Comment by ADmad** [comment]
By default the skeleton app use DashedRoute class for the router hence the URL that you should be using is `/underscore-tests`.

**Comment by GusAntoniassi** [comment]
Didn't know that, `/underscore-tests` works fine here. Still, should `/underscore_tests` be throwing a MissingControllerException? It is a little misleading, since the file *is* there.

**Comment by ADmad** [comment]
Yes the latter should be throwing an exception but perhaps we should improve the error message.

**Comment by markstory** [comment]
I took a look at this and because of the dashed route, the resulting controller parameter is `Underscore_tests` which is why the controller cannot be found. The MissingController error page, then uses Inflection to generate class & file names which is the source confusing error.

Would it be reasonable to generate a different error page content if the `controller` parameter contains an `_`? Instead of pointing to the file/class that is missing it can recommend checking your routes?

**Comment by GusAntoniassi** [comment]
Yes, that would be quite good. I would suggest to hint that the default skeleton route now uses dashes instead of underscores (something like "Did you mean: [underscore-tests](#)"); that'd be an useful tip for those coming from 2.x like me.
Also it could be worth adding a `Changed in version 3.x` in the [bake documentation page](https://book.cakephp.org/3.0/en/bake/usage.html) or the [migration guide](https://book.cakephp.org/3.0/en/appendices/3-0-migration-guide.html), since the default route was underscored in CakePHP 2.x, and those were the two pages I consulted when I found this error.
That information is however noted on the [CakePHP conventions page](https://book.cakephp.org/3.0/en/intro/conventions.html#url-considerations-for-controller-names), which I didn't think of looking into (shame on me).

Best regards.
